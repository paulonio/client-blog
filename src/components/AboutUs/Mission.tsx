import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';

import { MissionType } from '@/constants/constants';

import { sen } from '@/styles/fonts';
import styles from './styled.module.scss';

interface MissionProps {
  mission: MissionType;
}

const Mission: FC<MissionProps> = ({ mission }) => {
  const { rubric, title, body } = mission;
  const { t } = useTranslation('mission');

  return (
    <div className={styles.about__item}>
      <p className={styles.about__title}>{t('header')}</p>
      <div className={styles.about__content}>
        <h3 className={sen.className}>{t('title')}</h3>
        <p className="body-s">{t('body')}</p>
      </div>
    </div>
  );
};

export default Mission;
