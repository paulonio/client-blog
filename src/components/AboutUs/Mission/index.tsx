import React from 'react';
import { useTranslation } from 'next-i18next';

import { sen } from '@/styles/fonts';
import styles from './styled.module.scss';

const Mission = () => {
  const { t } = useTranslation('mission');

  return (
    <div className={styles.about__item}>
      <p className={styles.about__title}>{t('header')}</p>
      <div className={styles.about__content}>
        <h3 className={sen.className}>{t('title')}</h3>
        <p className="body-s">{t('body')}</p>
      </div>
    </div>
  );
};

export default Mission;
