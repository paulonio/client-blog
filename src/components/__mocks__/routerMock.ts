import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

(useRouter as jest.Mock).mockReturnValue({ query: {} });
