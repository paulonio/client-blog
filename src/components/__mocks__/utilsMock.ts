import { isLinkActive, parseString } from '@/utils/utils';

jest.mock('../../utils/utils', () => ({
  isLinkActive: jest.fn(),
  parseString: jest.fn(),
}));

(isLinkActive as jest.Mock).mockReturnValue(true);
(parseString as jest.Mock).mockReturnValue('');
