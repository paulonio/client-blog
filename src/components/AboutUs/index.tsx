'use client';

import React, { useRef } from 'react';
import Link from 'next/link';
import { useTranslation } from 'next-i18next';

import Pattern from './Pattern';
import Mission from './Mission';
import Button from '../Button';

import { useIntersectionObserver } from '@/hooks/useIntersectionObserver';

import { sen } from '@/styles/fonts';
import styles from './styled.module.scss';

const AboutUs = () => {
  const ref = useRef<HTMLElement | null>(null);
  const isIntersecting = useIntersectionObserver(ref);
  const { t } = useTranslation('about');

  return (
    <section ref={ref} className={styles['about-us']}>
      {isIntersecting && (
        <>
          <Pattern />
          <div className={styles.about__wrapper}>
            <div className={styles.about__item}>
              <p className={styles.about__title} data-testid="about-us-title">
                {t('header')}
              </p>
              <div className={styles.about__content}>
                <h2 className={sen.className}>{t('title')}</h2>
                <p className="body-s">{t('body')}</p>
                <Link href="/about">
                  <Button buttonType="link">
                    {t('read')} {'>'}
                  </Button>
                </Link>
              </div>
            </div>
            <Mission />
          </div>
        </>
      )}
    </section>
  );
};

export default AboutUs;
