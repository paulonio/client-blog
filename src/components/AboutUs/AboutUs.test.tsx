import React from 'react';
import { render } from '@testing-library/react';
import AboutUs from '.';
import '../__mocks__/intersectionObserverMock';

describe('About us', () => {
  it('should render about us page', () => {
    const { container } = render(<AboutUs />);

    expect(container).toMatchSnapshot();
  });
});
