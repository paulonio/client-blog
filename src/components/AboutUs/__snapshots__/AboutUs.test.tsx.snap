// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About us should render about us page 1`] = `
<div>
  <section
    class="about-us"
  />
</div>
`;
